### ISSUE TITLE
Warning: Model is duplicated! Please update the model version!

### ISSUE THREAD

**Issue by chrisw14** [imixs/imixs-workflow#537]
Starting Wildfly 17 I have several (about 300) warnings like this:

```
14:51:31,777 WARNING [org.imixs.workflow.engine.ModelService] (ServerService Thread Pool -- 77) Model 'Test' is dupplicated! Please update the model version!
14:51:31,795 INFO  [org.imixs.workflow.bpmn.BPMNParser] (ServerService Thread Pool -- 77) BPMN Model 'Mitarbeiter' parsed in 17ms
14:51:31,795 WARNING [org.imixs.workflow.engine.ModelService] (ServerService Thread Pool -- 77) Model 'Mitarbeiter' is dupplicated! Please update the model version!
14:51:31,817 INFO  [org.imixs.workflow.bpmn.BPMNParser] (ServerService Thread Pool -- 77) BPMN Model '1.0.2' parsed in 22ms
14:51:31,817 WARNING [org.imixs.workflow.engine.ModelService] (ServerService Thread Pool -- 77) Model '1.0.2' is dupplicated! Please update the model version!
14:51:31,832 INFO  [org.imixs.workflow.bpmn.BPMNParser] (ServerService Thread Pool -- 77) BPMN Model 'Mitarbeiter' parsed in 15ms
14:51:31,832 WARNING [org.imixs.workflow.engine.ModelService] (ServerService Thread Pool -- 77) Model 'Mitarbeiter' is dupplicated! Please update the model version!
14:51:31,835 INFO  [org.imixs.workflow.bpmn.BPMNParser] (ServerService Thread Pool -- 77) BPMN Model '1.0.0' parsed in 3ms
14:51:31,835 WARNING [org.imixs.workflow.engine.ModelService] (ServerService Thread Pool -- 77) Model '1.0.0' is dupplicated! Please update the model version!
```

Of course, each model exists only one time. But I don't know why I get these warnings?
In the init method of org.imixs.workflow.engine.ModelService I can see that it makes sense but why is the modelService initiated so often to parse the models?

**Comment by rsoika** [comment]
This indicates that you have a lot of Model-Document Entities stored in your database. The method ModelService.saveModel() takes care about such duplications and clean up the database automatically. 

How did you add the models in your application?

If you  search for :   

    (type:"model")

you should see that there are more than one model with the same model version.

**Comment by chrisw14** [comment]
I see some models, but always with a different txtname. The $version is always 1 because I always update them on server startup.

**Comment by chrisw14** [comment]
Found the mistake: I was using removeModel instead of deleteModel all the time.

**Comment by chrisw14** [comment]
... now I have these warnings again and after creating a new lucence index, I could found several model documents. But in my modelSetupService I'm using deleteModel(name) before adding the model on startup again (to update the model). Why do I have so much model documents?

**Comment by rsoika** [comment]
Yes I too think we have an issue here. But I do not yet understand why this happens....

**Comment by chrisw14** [comment]
Just a guess from my side: Is it possible that the model documents can't be deleted from database if they aren't indexed by Lucence? How can I start to rebuild the Lucence index from Java?
Is it really necessary to have this model documents in the lifecycle of the application? I can't find any disadvantage if I delete all model documents...

**Comment by rsoika** [comment]
I tried to reproduce the issue. But with the latest version 5.1.0 it was not possible any more.
We had fixed an issue in 5.1.0 concerning the model management

see #581 

In the past the effect you describe here was issued by an outdated lucene index. But with 5.1.0 we fixed this issue. 

So I believe the issue is no longer visible in version 5.1.0. Can you confirm this?
